Observer/Category.php

```diff
@@ -29,12 +28,18 @@ public function __construct(
     public function execute(Observer $observer)
     {
         try {
-            $category = $observer->getEvent()->getCategory();
-            if ($category->getStoreId() == 0) {
-                $categoryStoreIds = $this->helper->getStoreIdsPerProject($category->getStoreIds());
+            $category        = $observer->getEvent()->getCategory();
+            $categoryStoreId = $category->getStoreId();
+            
+            if ($categoryStoreId == 0) {
+                $categoryStoreIds   = $this->helper->getStoreIdsPerProject($category->getStoreIds());
             } else {
-                $categoryStoreIds[] = $category->getStoreId();
+                if (!$this->helper->isEnabled($categoryStoreId)) {
```

Comment:
The nesting of the `if` is gnarly here is there anything we can do? What do you think about moving the `isEnabled` check to the `foreach` on 43?

Reply:
Function isEnabled is a database ping and if i move it to the foreach i will create extra db interactions for each configured project. This was my consideration when nesting the if statement.